src/Services/GTT/shared/GTT.Infrastructure/Repositories/ChallengeRepository.cs

```diff
@@ -1,72 +1,58 @@
-        {
-            _connection = connection;
-            _tran = tran;
-        }
 
-        public async Task<int> AddAsync(Challenge entity)
+        public ChallengeRepository(IDbConnectionFactory dbConnectionFactory)
         {
-            var insertChallengeSql = @"
-                    INSERT INTO Challenge(field1, field2, field3)
-                    VALUES(@field1, @field2, @field3)
-                    SET @ChallengeId = SCOPE_IDENTITY()
-                    SELECT * FROM Challenge WHERE ChallengeId = @ChallengeId
-                ";
-
-            var param = new
-            {
-                field1 = "entity.field1",
-                field2 = "entity.field2",
-                field3 = "entity.field3"
-            };
-
-            var result = await _connection.ExecuteAsync(insertChallengeSql, param, transaction: _tran);
-
-            return result;
+            _connection = dbConnectionFactory.CreateConnection();
         }
-
-        public async Task<int> DeleteAsync(int id)
+        public ChallengeRepository(IDbConnection dbConnection) 
         {
-            var insertChallengeSql = @"DELETE FROM Products WHERE Id = @Id";
-            var result = await _connection.ExecuteAsync(insertChallengeSql, new { Id = id }, _tran);
-
-            return result;
+            _connection = dbConnection;
         }
 
-        public async Task<IReadOnlyList<Challenge>> GetAllAsync()
+        public async Task<ChallengeVM> AddAsync(CreateChallengeData challenge)
         {
-            var insertChallengeSql = @"SELECT * FROM Challenge";
-            var result = await _connection.QueryAsync<Challenge>(insertChallengeSql, _tran);
-
-            return result.ToList();
-        }
-
-        public async Task<Challenge> GetByIdAsync(int id)
-        {
-            var insertChallengeSql = @"SELECT * FROM Challenge WHERE Id = @Id";
-            var result = await _connection.QuerySingleOrDefaultAsync<Challenge>(insertChallengeSql, new { Id = id }, _tran);
-
-            return result;
-        }
-
-        public async Task<int> UpdateAsync(Challenge entity)
-        {
-            var sql = @"UPDATE Challenge SET something equal something";
-            var result = await _connection.ExecuteAsync(sql, entity, _tran);
-
-            return result;
+            try
+            {
+                var insertChallengeSql = @"INSERT INTO Challenge(Calories, SplatPoints, AvgHR, MaxHR, Miles, Steps, MemberID, ClassID, CreatedDate, UpdatedDate, CreatedBy, UpdatedBy)
+                                    VALUES(@Calories, @SplatPoints, @AvgHR, @MaxHR, @Miles, @Steps, @MemberID, @ClassID, @CreatedDate, @UpdatedDate, @CreatedBy, @UpdatedBy)
+                                    DECLARE @challengeID int
+                                    SET @challengeID = SCOPE_IDENTITY()
+                                    SELECT* FROM Challenge WHERE challengeID = @ChallengeId
+                                    ";
+
```

Comment:
We should check Class and Member whether both exit in another table before adding one challenge in two table